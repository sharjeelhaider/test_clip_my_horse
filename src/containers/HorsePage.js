import {connect} from 'react-redux';
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {SafeAreaView, StyleSheet, Text, View} from 'react-native';
import HorseList from '../components/HorseList';

// Exporting for testing purposes
export class HorsePage extends Component {
  render() {
    const {horses} = this.props.horses;
    return (
      <SafeAreaView style={styles.container}>
        <View style={styles.horseContainer}>
          <Text style={styles.title}>Horses</Text>
          <View style={styles.divider} />
          <HorseList horses={horses} />
        </View>
      </SafeAreaView>
    );
  }
}

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'black',
  },
  horseContainer: {
    flex: 1,
  },
  divider: {
    backgroundColor: 'grey',
    height: 4,
  },
  title: {
    fontSize: 22,
    fontWeight: 'bold',
    alignSelf: 'center',
    backgroundColor: 'black',
    padding: 15,
    color: 'red',
  },
});

HorsePage.propTypes = {
  horses: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({horses: state.horses});

export default connect(mapStateToProps)(HorsePage);
