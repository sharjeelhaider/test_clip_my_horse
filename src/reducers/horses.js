import {LOAD_HORSES} from '../actions/api';

export const initialState = {
  // Exporting it for test purposes
  error: null,
  isLoading: false,
  horses: null,
  isLoggedIn: false,
};

export default function horses(state = initialState, action) {
  switch (action.type) {
    case LOAD_HORSES.failure:
      return {...state, error: action.error, isLoading: false};
    case LOAD_HORSES.request:
      return {...state, error: null, isLoading: true};
    case LOAD_HORSES.success:
      return {
        ...state,
        error: null,
        isLoading: false,
        isLoggedIn: true,
        horses: action.data,
      };
    default:
      return state;
  }
}
