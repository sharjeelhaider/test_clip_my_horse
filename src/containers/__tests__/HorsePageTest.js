import React from 'react';
import {HorsePage} from '../HorsePage';
import horses from '../../../config/jest/dataMock';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('../../components/HorseList', () => 'HorseList');

it('renders a HorsePage using Snapshots', () => {
  const _getComponent = props => {
    return renderer.create(<HorsePage horses={props} />);
  };

  let component = _getComponent({horses: []});
  expect(component).toMatchSnapshot();

  component = _getComponent({horses: horses});
  expect(component).toMatchSnapshot();
});
