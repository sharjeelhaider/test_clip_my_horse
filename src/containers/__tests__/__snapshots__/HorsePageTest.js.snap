// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a HorsePage using Snapshots 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "color": "red",
          "fontSize": 22,
          "fontWeight": "bold",
          "padding": 15,
        }
      }
    >
      Horses
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "grey",
          "height": 4,
        }
      }
    />
    <HorseList
      horses={Array []}
    />
  </View>
</RCTSafeAreaView>
`;

exports[`renders a HorsePage using Snapshots 2`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "color": "red",
          "fontSize": 22,
          "fontWeight": "bold",
          "padding": 15,
        }
      }
    >
      Horses
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "grey",
          "height": 4,
        }
      }
    />
    <HorseList
      horses={
        Array [
          Object {
            "id": "df579a48-a7ca-4afc-89d0-caabcfac87d7",
            "name": "American Beauty",
            "picture": "https://d2vtwgzbrv2xnd.cloudfront.net/51e87826c943fd896385734287bbac27fd43c74a.jpeg",
          },
          Object {
            "id": "8b1b98fe-077f-4faf-bb16-f088f3b6dfdf",
            "name": "Arabian Night",
            "picture": "https://d2vtwgzbrv2xnd.cloudfront.net/dae8d44c97b286e627a62569c2ff46bbb4575270.jpeg",
          },
          Object {
            "id": "49092d28-2d7b-4774-8a40-7c1a9d3e7634",
            "name": "Autumn Gold",
            "picture": "https://d2vtwgzbrv2xnd.cloudfront.net/b05678d2169739f280993fb8636ba017ff9b6787.jpeg",
          },
          Object {
            "id": "be11640d-1fe6-4d73-a664-0fd84ea6338c",
            "name": "Ancient Orange",
            "picture": "https://d2vtwgzbrv2xnd.cloudfront.net/5ea73acc5924c18bec341e3b30eb6ba8ea463d28.jpeg",
          },
          Object {
            "id": "a04e73cf-6ceb-40e9-969e-e94b02f4941e",
            "name": "Amazing Grace",
            "picture": "https://d2vtwgzbrv2xnd.cloudfront.net/7d860756e5c7fc9df00b39b459b868a12edbac9a.jpeg",
          },
          Object {
            "id": "a3c4c44c-1975-4d6d-a870-af9cd5d23628",
            "name": "Apple Candy",
            "picture": "https://d2vtwgzbrv2xnd.cloudfront.net/a7acc104b765cf17c4adeabaa2ca401ace08a5df.jpeg",
          },
        ]
      }
    />
  </View>
</RCTSafeAreaView>
`;
